var/spack/repos/builtin/packages/snakemake/package.py

```diff
@@ -53,3 +59,18 @@ class Snakemake(PythonPackage):
         depends_on("py-google-crc32c", type=("build", "run"))
         depends_on("py-google-api-python-client", type=("build", "run"))
         depends_on("py-google-cloud-storage", type=("build", "run"))
+
+    # These variants are not in PyPI/pip, but they are undocumented dependencies
+    # needed to make certain parts of Snakemake work.
+    variant("ftp", default=False, description="Enable snakemake.remote.FTP")
+    depends_on("py-ftputil", when="+ftp")
+
+    variant("s3", default=False, description="Enable snakemake.remote.S3")
+    depends_on("py-boto3", when="+s3")
+    depends_on("py-botocore", when="+s3")
+
+    variant("http", default=False, description="Enable snakemake.remote.HTTP")
+    depends_on("py-requests", when="+http")
```

Comment:
These should probably be `type="run"` or `type=("build", "run")` so that they are available when you load the package.